CodePush.js

```diff
@@ -75,46 +78,23 @@ async function checkForUpdate(deploymentKey = null, handleBinaryVersionMismatchC
            * @type {string}
            */
           const runtimeVersion = sharedCodePushOptions.runtimeVersion;
+          const Versioning = sharedCodePushOptions.versioning;
 
-
-          // NOTE: It is not necessary to implement it in three steps. However, it is recommended to implement unit-test-friendly.
-          /**
-           * TODO 1: find latest release in releaseHistory
-           * @return {ReleaseInfo}
-           */
-          function findLatestRelease(releaseHistory) {}
-
-          /**
-           * TODO 2: check if the update is mandatory
-           * @return {boolean}
-           */
-          function checkIsMandatory(runtimeVersion, releaseHistory) {}
-
-          /**
-           * TODO 3: determine whether to rollback and execute it
-           * @return {boolean}
-           */
-          function shouldRollback(runtimeVersion, releaseHistory) {}
-          if (shouldRollback(runtimeVersion, releaseHistory)) {
-            // rollback
-          }
-
-          // NOTE: sample code
-          const update = findLatestRelease(releaseHistory);
-          const isMandatory = checkIsMandatory(runtimeVersion, update);
-
+          const [latestReleaseVersion ,latestReleaseInfo] = Versioning.findLatestRelease(releaseHistory);
+          const isMandatory = Versioning.checkIsMandatory(runtimeVersion, latestReleaseVersion);
+          const shouldRollback = Versioning.shouldRollback(runtimeVersion, releaseHistory)
```

Comment:
@changwoolab 
we should implement the 'rollback to binary-version' behavior.
(clearing current bundle and restart)

Reply:
@floydkim 
Included in [5305269](https://github.com/Soomgo-Mobile/react-native-code-push/pull/27/commits/5305269d1d6891614b10ba8b49e9ac97323600f8)